### PR TITLE
Add tips for version extraction

### LINKED ISSUE #1033: Add tips on extracting version information from commands
**Is your feature request related to a problem? Please describe.**
Complicated solutions are often presented for extracting version numbers.
e.g.
```
echo $(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//'
```

**Describe the solution you'd like**
Add some information on efficient ways to get the version number to the adding modules page:
e.g.
```
samtools --version | sed '1!d; s/samtools //'
```
Tips:
- `sed '3!d'` Extracts only line 3 of the output printed by `samtools --version`.
- Determine whether the error printed to stderr or stdout, by trying to filter the line with `sed`
    ```
    samtools --version
    # Try filtering the specific line
    samtools --version | sed '1!d'
    ```
    If it works, then you're reading from stdout, otherwise you need to capture stderr using `|&` which is shorthand for
    `2>&1 |`.
- You can separate sed commands using `;`. Often the pattern - filter line; then replace string is enough to get the version number.
- `sed 's/pattern/replacement/'` can be used to remove parts of a string. `.` matches any character, `+` matches 1 or more times.
- It is not necessary to use `echo`.
- What's a solution for non-zero error code? `command --version || true` but what about ` command --version | sed ... || true`?

### PR DESCRIPTION
**Pull request by LouisLeNezet**
fix #1033 

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1722"></a>

### CODE REVIEW

File: markdown/developers/tutorials/dsl2_modules_tutorial.md [hunk 1]
```diff
@@ -190,8 +192,35 @@ In order to avoid that, we can in general print the version as part of an echo s
...
+
+      ```bash
+      samtools --version
+      # Try filtering the specific line
+      samtools --version | sed '1!d'
+      ```
+
+      - If it works, then you're reading from stdout, otherwise you need to capture stderr using `|&` which is shorthand for `2>&1 |`
+      - You can separate sed commands using `;`. Often the pattern : `sed filter line ; replace string` is enough to get the version number
+      - `sed 's/pattern/replacement/'` can be used to remove parts of a string. `.` matches any character, `+` matches 1 or more times.
```
Comment (mashehu) [inline markdown/developers/tutorials/dsl2_modules_tutorial.md:216]:
```suggestion
      - `sed 's/pattern/replacement/'` can be used to remove parts of a string. `.` matches any character, `+` matches 1 or more times.
      - You can separate sed commands using `;`. Often the pattern : `sed filter line ; replace string` is enough to get the version number
```

File: markdown/developers/tutorials/dsl2_modules_tutorial.md [hunk 2]
```diff
@@ -190,8 +192,35 @@ In order to avoid that, we can in general print the version as part of an echo s
...
+      - It is not necessary to use `echo`
+      - For non-zero error code: `command --version || true` or  `command --version | sed ... || true`
+      - If the version is at a specific line you can try `sed -nr '/pattern/p'` that will return only the line with the pattern
+      - To extract the version number in the middle you can also use regex pattern with `grep` as follow: `grep -o -E '([0-9]+.){1,2}[0-9]'`
+      - If multiple lines are returned you can select the first one with `head -n 1`
```
Comment (mashehu) [inline markdown/developers/tutorials/dsl2_modules_tutorial.md:221]:
```suggestion
      - If multiple lines are returned you can select the first one with `tool --version | head -n 1`
```

**Review by mashehu — APPROVED** [review]

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
Agree, very helpfuyl! Thanks @LouisLeNezet !

**Review by lescai — APPROVED** [review]
I really don't have any more comments to add, it looks very useful and thanks for adding this part in so much detail!

**Pull request merged**